Snowflake.Data.Tests/UnitTests/SFGCSClientTest.cs

```diff
@@ -0,0 +1,457 @@
+﻿/*
+ * Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+namespace Snowflake.Data.Tests.UnitTests
+{
+    using NUnit.Framework;
+    using Snowflake.Data.Core;
+    using Snowflake.Data.Core.FileTransfer.StorageClient;
+    using Snowflake.Data.Core.FileTransfer;
+    using System.Collections.Generic;
+    using System.Net;
+    using System.IO;
+    using System.Threading.Tasks;
+    using System.Threading;
+    using Snowflake.Data.Tests.Mock;
+    using Moq;
+    using System.Text;
+
+    [TestFixture]
+    class SFGCSClientTest : SFBaseTest
+    {
+        // Mock data for file metadata
+        const string LocationStage = "mock-customer-stage";
+        const string LocationId = "mock-id";
+        const string LocationTables = "tables";
+        const string LocationKey = "mock-key";
+        const string LocationPath = LocationTables + "/" + LocationKey + "/";
+        const string Location = LocationStage + "/" + LocationId + "/" + LocationPath;
+
+        const string Region = "US-CENTRAL1";
+
+        Dictionary<string, string> _stageCredentials = new Dictionary<string, string>()
+        {
+            {"GCS_ACCESS_TOKEN", "MOCK_GCS_ACCESS_TOKEN"}
+        };
+
+        // Settings for mock client
+        const int Parallel = 0;
+
+        // File name for download tests
+        const string DownloadFileName = "mockFileName.txt";
+
+        // Token for async tests
+        CancellationToken _cancellationToken;
+
+        // Mock upload file size
+        const int UploadFileSize = 9999;
+
+        // The mock client and metadata
+        SFGCSClient _client;
+        SFFileMetadata _fileMetadata;
+
+        [SetUp]
+        public void BeforeTest()
+        {
+            _fileMetadata = new SFFileMetadata()
+            {
+                stageInfo = new PutGetStageInfo()
+                {
+                    endPoint = null,
+                    isClientSideEncrypted = true,
+                    location = Location,
+                    locationType = SFRemoteStorageUtil.GCS_FS,
+                    path = LocationPath,
+                    presignedUrl = null,
+                    region = Region,
+                    stageCredentials = _stageCredentials,
+                    storageAccount = null
+                }
+            };
+
+            // Setup mock GCS HTTP client
+            _client = new SFGCSClient(_fileMetadata.stageInfo);
+
+            _cancellationToken = new CancellationToken();
+        }
+
+        [Test]
+        [Ignore("GCSClientTest")]
+        public void GCSClientTestDone()
+        {
+            // Do nothing;
+        }
+
+        [Test]
+        public void TestConstructorWithoutGCSAccessToken()
+        {
+            _fileMetadata.stageInfo.stageCredentials = new Dictionary<string, string>();
+            new SFGCSClient(_fileMetadata.stageInfo);
```

Comment:
checked on the constructor and I would add at least Assert.NotNull or Assert.DoesNotThrow if that's the case here

Reply:
You mentioned other integration tests are already testing this so I'll just remove it